website/blog/2023-03-23-audit-helper.md

```diff
@@ -0,0 +1,253 @@
+
+With `compare_column_values` macro, the report output addresses specifically that problem, pointing out to the analyst which are the inconsistencies that may be causing low compatibility.
+
+### The code for the compare_column_values macro: Step-by-step
+1. Create a new .sql model in the folder of your choice in your dbt project
+2. Copy and paste the following example below in the model created:
+
+```sql
+{# in dbt Develop #}
+
+{% set old_etl_relation_query %}
+select * from public.dim_product
+where is_latest
+{% endset %}
+
+
+{% set new_etl_relation_query %}
+select * from {{ ref('dim_product') }}
+{% endset %}
+
+
+{% set audit_query = audit_helper.compare_column_values(
+    a_query=old_etl_relation_query,
+    b_query=new_etl_relation_query,
+    primary_key="product_id",
+    column_to_compare="status"
+) %}
+
+
+{% set audit_results = run_query(audit_query) %}
+
+
+{% if execute %}
+    {% do audit_results.print_table() %}
+{% endif %}
+```
+  
+The arguments used by this macro are pretty much the same used by the `compare_queries` macro, but with the addition of the `column_to_compare` argument, being that responsible, as the name suggests, for declaring which column is specifically going to be tested.
+
+Although similar to the compare_queries macro, this `compare_columns_value` macro gives great importance to the `primary_key` argument, which is going to be a fundamental part of generating correctly the metrics attributed to the compared column, acting as an anchor to compare every generated row from the refactored model with its legacy counterpart.
+
+Also, we can see that there is a table printing option enabled as default, so the result of the query gets printed on the terminal when the macro is run. That can be useful to quickly check out the compatibility status of a column, without leaving the code editor while refactoring SQL through dbt.
+
+3. Replace column names and source references from the example with the respective information of the compared models.
+4. Optional: Disable the `print_table()` command, so the model can be materialized on your data warehouse.
+    To disable table printing on terminal and to enable model materialization in your target data warehouse, the following can be done: replace this entire section of code, which declares the SQL model as variable and makes it printable, with the macro execution pattern.
+
+    ```sql
+    - Replace the commented code below:
+    - {% set audit_results = run_query(audit_query) %}
+
+    - {% if execute %}
+    - {% do audit_results.print_table() %}
+    - {% endif %}
+
+    - With the following piece of code:
+    {{ run_query(audit_query) }}
```

Comment:
I would recommend not using run_query at all! All we need to do is print the query here, so  `{{ audit_query }}` should be sufficient